### PR TITLE
OCPBUGS-25115: Fix Racing Machine Configs and add Day 0 Support (#854)

### PR DESCRIPTION
**Pull request by rbaturov**
This is a manual cherry pick of "Fix Racing Machine Configs and add Day 0 Support"(https://github.com/openshift/cluster-node-tuning-operator/pull/854)

### CODE REVIEW

**Comment by jmencak** [comment]
This rings some bells
/cc @ffromani

**Comment by yanirq** [comment]
@jmencak This is a manual cherry pick of https://github.com/openshift/cluster-node-tuning-operator/pull/854

**Comment by Tal-or** [comment]
```
 STEP: Getting MCP for profile @ 12/11/23 16:06:38.929
  [FAILED] in [It] - /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/0_config/config.go:89 @ 12/11/23 16:07:44.074
• [FAILED] [275.746 seconds]
[performance][config] Performance configuration [It] Should successfully deploy the performance profile
/go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/0_config/config.go:66
  [FAILED] Failed getting MCP by label key machineconfiguration.openshift.io/role value worker-cnf
  Unexpected error:
      <*errors.StatusError | 0xc0005343c0>: 
      the server was unable to return a response in the time allotted, but may still be processing the request (get machineconfigpools.machineconfiguration.openshift.io)
      {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "the server was unable to return a response in the time allotted, but may still be processing the request (get machineconfigpools.machineconfiguration.openshift.io)",
              Reason: "Timeout",
              Details: {
                  Name: "",
                  Group: "machineconfiguration.openshift.io",
                  Kind: "machineconfigpools",
                  UID: "",
                  Causes: [
                      {
                          Type: "UnexpectedServerResponse",
                          Message: "{\"metadata\":{},\"status\":\"Failure\",\"message\":\"Timeout: request did not complete within the allotted timeout\",\"reason\":\"Timeout\",\"details\":{},\"code\":504}",
                          Field: "",
                      },
                  ],
                  RetryAfterSeconds: 0,
              },
              Code: 504,
          },
      }
  occurred
  In [It] at: /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/0_config/config.go:89 @ 12/11/23 16:07:44.074
```
Doesn't seems familiar

File: test/e2e/performanceprofile/functests/1_performance/performance.go [hunk 1]
```diff
@@ -445,105 +445,6 @@ var _ = Describe("[rfe_id:27368][performance]", Ordered, func() {
 		})
 	})
 
-	Context("KubeletConfig experimental annotation", func() {
```
Comment (MarSik) [inline test/e2e/performanceprofile/functests/1_performance/performance.go:448]: @ffromani Do you remember why we removed this? I see it is part of the original patch too.

Comment (ffromani) [reply test/e2e/performanceprofile/functests/1_performance/performance.go:448]: yes: https://github.com/openshift/cluster-node-tuning-operator/pull/854/commits/a623c50e0ea8dd322a7b319731eccde68f966678

**Comment by ffromani** [comment]
LGTM

would like a second review/have another look later/both

**Comment by ffromani** [comment]
/approve

in general, we do want this backport

File: test/e2e/performanceprofile/functests/1_performance/performance.go (hunk 1 above)
Comment (MarSik) [reply test/e2e/performanceprofile/functests/1_performance/performance.go:448]: +1

**Pull request merged**